src/main/scala/mesosphere/marathon/api/v2/validation/AppValidation.scala

```diff
@@ -349,8 +349,14 @@ trait AppValidation {
     app must requireUnreachableDisabledForResidentTasks
     app.constraints.each must complyWithAppConstraintRules
     app.networks is ramlNetworksValidator
+    app is validWithRoleEnforcement(validRoles)
```

Comment:
I could not find another faster way to validate the app role. We have validations for `raml.App` and `state.AppDefinition`. If you accept this diff I promise I'll follow up and remove one validation.